app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -60,6 +62,11 @@ def deployment_spec(opts)
     deploy_options
   end
 
+  def workflow_helper
+    values = {:src_vm_id => [id], :src_vm_class_name => "OrchestrationTemplate"}
```

Comment:
A little concerned that passing in something other than a VM to something that has been expecting a VM for a decade might cause some unexpected issues

Reply:
It sounds risky 😄

Reply:
Is it good to make it type of `VmOrTemplate`?

Reply:
Well I'm worried that if the ProvisionWorkflow is expecting a VM in `src_vm` and we're passing in something else, we're going to have to deal with all method calls, e.g. is https://github.com/ManageIQ/manageiq/pull/20636/files#diff-1ce4751059050fe9933b971da4bb02ddR412 really the _only_ time the provision workflow calls a method on src_vm?

Reply:
cc @Fryguy

Reply:
That is the case with 3 `allowed_` methods for resource pools, hosts and Ems_folders.
I'm going to check the other `allowed_` methods.

Reply:
Tested with all `allowed_` methods in `miq_provision_dialogs.yaml`.  Only hit an issue with `allowed_snapshots` which has been addressed in ManageIQ/manageiq#20636.

Reply:
Okay, @lfu can you add specs to cover these cases with an OrchestrationTemplate as a src_vm?  I know I wouldn't think twice about assuming src_vm is a Vm object so specs at least are critical.

Longer term maybe we should look at making `ProvisionWorkflow` either more general or just changing `src_vm` to `src_obj` or something